src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -956,10 +977,20 @@ class OpenShiftDSL implements Serializable {
             if ( objectList != null ) {
                 return objectList;
             }
-
+            
             // Otherwise, we need to ask the API server what presently matches
-            OcAction.OcActionResult r = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("get", selectionArgs(), null, "-o=name") );
-            r.failIf( "Unable to retrieve object names: " + this.toString() );
+            OcAction.OcActionResult r = null;
+            ArrayList<String> g = new ArrayList<String>();
+            g.add("get")
+            OcAction.OcActionResult versionCheck = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("help", g, null, null) );
```

Comment:
The suggestion I included should re-acquire the capabilities of a client every time a pipeline runs (and so should cover the case of the tool configuration changes from run to run). It will also acquire it for every different node (in case you have heterogeneous client installs).